types/testingutils/verifier.go

```diff
@@ -17,21 +17,19 @@ type testingVerifier struct {
 }
 
 var (
-	testingVerifierInstance     *testingVerifier
-	testingVerifierInstanceLock sync.Mutex
+	testingVerifierInstance             *testingVerifier
+	testingVerifierInstanceLock         sync.Mutex
+	testingVerifierSingletonConstructor sync.Once
 )
 
 func NewTestingVerifier() types.SignatureVerifier {
-	if testingVerifierInstance == nil {
-		testingVerifierInstanceLock.Lock()
-		defer testingVerifierInstanceLock.Unlock()
-		if testingVerifierInstance == nil {
-			testingVerifierInstance = &testingVerifier{
-				signaturesCache: make(map[types.OperatorID]map[[32]byte][]byte),
-			}
-			return testingVerifierInstance
+	testingVerifierInstanceLock.Lock()
+	defer testingVerifierInstanceLock.Unlock()
```

Comment:
no need for this lock I think now :-)

Reply:
that's the whole point of the `sync.Once` primitive in go. To solve this synchronization problem correctly so you don't add locks yourself.
There are issues in the memory model in some programming languages to do efficient double locking (including go), what you initially tried to do and then fixed by making less efficient. 

`sync.Once` solves this efficiently and elegantly

Reply:
Okk, great!!